src/aks-preview/azext_aks_preview/agentpool_decorator.py

```diff
@@ -383,6 +383,25 @@ def get_enable_artifact_streaming(self) -> bool:
             ):
                 enable_artifact_streaming = self.agentpool.artifact_streaming_profile.enabled
         return enable_artifact_streaming
+    
+    def get_pod_ip_allocation_mode(self, read_only: bool = False) -> Union[str, None]:
+        """Get the value of pod_ip_allocation_mode.
+        :return: str or None
+        """
+        # overwrite if provided by user
+        pod_ip_allocation_mode = self.raw_param.get("pod_ip_allocation_mode")
+        # In create mode, try to read the property value corresponding to the parameter from the `agentpool` object
+        # if the agentpool property is specified then the property value will be used and cannot be mutated
+        if self.decorator_mode == DecoratorMode.CREATE:
```

Comment:
you probably copy-pasta'd this from another property so not your fault but I think the correct thing is to use the raw_param always and if it isn't provided then read it from the agentpool

Reply:
Changed the logic here to initialize with agentpool value and override it with the raw value if it is provided and removed the decorator mode.